feature_server/server.py

```diff
@@ -324,13 +324,17 @@ def on_team_join(self, team):
                     return False
         if team.locked:
             self.send_chat('Team is locked')
+            if not team.spectator and not team.other.locked:
```

Comment:
seems fine. consider this: you selecting blue but it is locked. it will throw you to green